src/types.ts

```diff
@@ -25,7 +25,7 @@ export interface SignToolOptions extends OptionalSignToolOptions, OptionalHookOp
 
 export interface InternalSignOptions extends SignOptionsForFiles {}
 
-export interface InternalSignToolOptions extends OptionalSignToolOptions, OptionalHookOptions {
+export interface InternalSignToolOptions extends OptionalSignToolOptions, Omit<OptionalHookOptions, 'hashes'> {
```

Comment:
We're passing it anyway, so we might as well keep the easy-to-grok extension around. In other words, I'd just keep the type as is and include `hashes`, even if it's unused.

Reply:
(I was trying to impact as few bits as possible, hence the `Omit<>`)

Reply:
BTW. The `hashes` are not included in `OptionalHookOptions`. The `hashes` property is in `OptionalSignToolOptions`.